eventuals/notification.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "eventuals/lock.h"
+#include "eventuals/then.h"
+
+////////////////////////////////////////////////////////////////////////
+
+namespace eventuals {
+
+////////////////////////////////////////////////////////////////////////
+
+// This header file defines a 'Notification' abstraction, which allows
+// an eventual to be able to wait for a single occurrence of a single
+// event. Said another way, the 'Notification' object maintains a
+// private boolean "notified" that transitions to 'true' at most once.
+//
+// This is similar to 'absl::Notification'
+// (https://abseil.io/docs/cpp/guides/synchronization#notification).
+//
+// You can call 'Wait()' on a 'Notification' to wait until the
+// "notified" state is 'true' and 'Notify()' to set the notification's
+// "notified" state to 'true' and notify all waiting eventuals that
+// the event has occurred. This method should only be called once and
+// will do nothing on repeated calls.
+//
+// Note that while 'Notify()' should only be called once, it is
+// perfectly valid to call 'Wait()' multiple times and/or from
+// multiple eventuals -- even after the notification's "notified"
+// state has been set -- in which case those calls will not wait.
+//
+// Note that the lifetime of a 'Notification' requires careful
+// consideration; it might not be safe to destroy a 'Notification'
+// after calling 'Notify()' since other eventuals may have called, or
+// will call, 'Wait()'.
+class Notification final : public Synchronizable {
+ public:
+  Notification()
+    : notification_(&lock()) {}
+
+  ~Notification() override = default;
+
+  [[nodiscard]] auto Notify() {
```

Comment:
Please add comments on these public methods.

We can get good inspiration from https://github.com/abseil/abseil-cpp/blob/149ab40581d33febf3f8bddb52505c9ee9edd838/absl/synchronization/notification.h#L103-L108 :

Here's my attempt:
```
// Sets the notified state of this notification to true and wakes waiting threads.
// Should only be called once. Repeated calls are undefined behavior.
```